File: examples/ruby/search_2.rb
def binary_search(arr, target)
  lo, hi = 0, arr.size - 1
  while lo <= hi
    mid = (lo + hi) / 2
    return mid if arr[mid] == target
    arr[mid] < target ? lo = mid + 1 : hi = mid - 1
  end
  -1
end

data = [1, 3, 5, 7, 9]
puts binary_search(data, 5)
puts binary_search(data, 4)

File: examples/python/search_2.py
def binary_search(arr, target):
    lo, hi = 0, len(arr) - 1
    while lo <= hi:
        mid = (lo + hi) // 2
        if arr[mid] == target:
            return mid
        if arr[mid] < target:
            lo = mid + 1
        else:
            hi = mid - 1
    return -1


if __name__ == "__main__":
    data = [1, 3, 5, 7, 9]
    print(binary_search(data, 5))
    print(binary_search(data, 4))
